tests/core/api/GH_ORG_SUBS_API.js

```diff
@@ -0,0 +1,436 @@
+
+var testSetup = require('../../../testSetup.js');
+var backoff = require('backoff');
+
+var testSuite = 'GH_ORG_SUB_API';
+var testSuiteDesc = 'Github Organization subscription API tests';
+var test = util.format('%s - %s', testSuite, testSuiteDesc);
+
+describe(test,
+  function () {
+    var ownerApiAdapter = null;
+    var collaboraterApiAdapter = null;
+    var memberApiAdapter = null;
+    var unauthorizedApiAdapter = null;
+    var subscriptions = [];
+    var subscription = {};
+    var runId = null;
+    var project = {};
+    var successStatusCode = null;
+    var runStatus = [];
+
+    this.timeout(0);
+    before(
+      function (done) {
+        async.series(
+          [
+            testSetup.bind(null)
+          ],
+          function (err) {
+            if (err) {
+              logger.error(test, 'Failed to setup tests. err:', err);
+              return done(err);
+            }
+            ownerApiAdapter =
+              global.newApiAdapterByStateAccount('ghOwnerAccount');
+            collaboraterApiAdapter =
+              global.newApiAdapterByStateAccount('ghCollaboratorAccount');
+            memberApiAdapter =
+              global.newApiAdapterByStateAccount('ghMemberAccount');
+            unauthorizedApiAdapter =
+              global.newApiAdapterByStateAccount('ghUnauthorizedAccount');
+
+            successStatusCode = _.findWhere(global.systemCodes,
+              {group: 'statusCodes', name: 'SUCCESS'}).code;
+
+            ownerApiAdapter.getProjects('',
+              function (err, prjs) {
+                if (err || _.isEmpty(prjs))
+                  return done(new Error('Project list is empty', err));
+
+                project = _.first(
+                  _.where(prjs, {isOrg: true, isPrivateRepository: true}
+                  )
+                );
+
+                project.test_resource_type = 'project';
+                project.test_resource_name = 'ghOrgPrivate';
+                assert.isNotEmpty(project, 'User cannot find the project');
+                return done();
+              }
+            );
+          }
+        );
+      }
+    );
+
+    it('1. Owner can get all its subscriptions',
+      function (done) {
+        ownerApiAdapter.getSubscriptions('',
+          function (err, subs) {
+            if (err || _.isEmpty(subs))
+              return done(
+                new Error(
+                  util.format('User cannot get subscription',
+                    query, err)
+                )
+              );
+            subscriptions = subs;
+            assert.isNotEmpty(subscriptions, 'User cannot find the subscriptions');
+            assert.equal(subscriptions.length,
+              global.ADM_GH_SUB_COUNT, 'Subscription count needs to match');
+            return done();
+          }
+        );
+      }
+    );
+
+    it('2. Owner can get subscription by ID',
+      function (done) {
+        var sub = _.findWhere(subscriptions, {orgName:global.TEST_GH_ORGNAME}) || {};
```

Comment:
why do we need to do this? why not just use the project that you setup, use its subscriptionId